packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -279,7 +369,9 @@ abstract class Route<T> {
   /// rendered. It is even possible for the route to be active but for the stateful
   /// widgets within the route to not be instantiated. See [ModalRoute.maintainState].
   bool get isActive {
-    return _navigator != null && _navigator._history.contains(this);
+    if (_navigator == null)
+      return false;
+    return _navigator._history.firstWhere(_RouteEntry.isRoutePredicate(this)).isPresent;
```

Comment:
This will now throw `NoElementException` on failure - is that intended?

Reply:
If the route has a navigator it should also be in the history unless you call isActive from dispose, I guess. You shouldn't do that.... I made this check stronger to even handle that case.

Reply:
It might be overly defensive, but maybe we could at least throw an exception other than NoSuchElement so that a user would know why this crashed?

Reply:
(Sorry didn't push yet, I changed it so the exception can never happen)